api/v3/Case/Getdetails.php

```diff
@@ -52,10 +58,21 @@ function civicrm_api3_case_getdetails($params) {
     if (!is_array($params['case_manager'])) {
       $params['case_manager'] = array('=' => $params['case_manager']);
     }
-    \Civi\CCase\Utils::joinOnManager($sql);
+    \Civi\CCase\Utils::joinOnRelationship($sql, 'manager');
     $sql->where(CRM_Core_DAO::createSQLFilter('manager.id', $params['case_manager']));
   }
 
+  // Add clause to search by non manager role and non client
+  if (!empty($params['contact_involved'])) {
+    if (!is_array($params['contact_involved'])) {
+      $params['contact_involved'] = array('=' => $params['contact_involved']);
+    }
+    \Civi\CCase\Utils::joinOnRelationship($sql, 'all');
+    $caseClient = CRM_Core_DAO::createSQLFilter('contact_id', $params['contact_involved']);
+    $nonCaseClient = CRM_Core_DAO::createSQLFilter('manager.id', $params['contact_involved']);
+    $sql->where("a.id IN (SELECT case_id FROM civicrm_case_contact WHERE ($nonCaseClient OR $caseClient))");
+  }
```

Comment:
this `civicrm_api3_case_getdetails ` function is _massive_. At the very least can you place this block of new code into a separate, documented function
```php
function _civicrm_api3_case_add_contact_involved_clause($params, $sql) {
  // ...
}
```